bots/github-push

```diff
@@ -0,0 +1,125 @@
+#! /usr/bin/python3
+
+# github-push -- Push the local branch to a GitHub PR, with benefits
+#
+# This tool pushes your local changes to a PR and also adds a comment
+# that summarizes what has changed.  The idea is to cement the actual
+# history of the PR in its comments, regardless of rebases, rewrites,
+# squashed fixups, etc.
+#
+# The tool finds the pull request number by looking through all open
+# pull requests on GitHub.  Once it has been found, the number is
+# cached in the Git config file to speed up subsequent invocations.
+
+import subprocess
+import argparse
+import sys
+import time
+
+from task import github
+
+def execute(*args):
+    try:
+        output = subprocess.check_output(args, universal_newlines=True)
+    except subprocess.CalledProcessError as ex:
+        sys.exit(ex.returncode)
+    return output
+
+def git(*args):
+    return execute("git", *args)
+
+def git_config(name):
+    try:
+        output = subprocess.check_output([ "git", "config", name ], universal_newlines=True)
+    except subprocess.CalledProcessError as ex:
+        return None
+    return output.strip()
+
+def main():
+    parser = argparse.ArgumentParser(description='Force push after a rewrite')
+    parser.add_argument('--repo', help="The GitHub repository to work with", default="cockpit-project/cockpit")
+    opts = parser.parse_args()
+
+    api = github.GitHub(repo=opts.repo)
+
+    local = git("rev-parse", "HEAD").strip()
+    remote = git("rev-parse", "@{u}").strip()
+
+    if local == remote:
+        sys.stderr.write("Nothing to push.\n")
+        return 0
+
+    ( tracking_remote, tracking_ref ) = git("rev-parse", "--abbrev-ref", "--symbolic-full-name", "@{u}").strip().split("/")
```

Comment:
Just to understand this: This is supposed to result in something like `(myremotename, branchname)`, right? If I run this on my `master` checkout, I get `(origin, master)`. (As said above, it doesn't work for me on branches)

Reply:
Can you show the relevant section from your `.git/config`?  Mine looks like this:

```
[branch "bots-github-push-tool"]
	remote = mvollmer
	merge = refs/heads/bots-github-push-tool
```

Reply:
> This is supposed to result in something like (myremotename, branchname), right? 

Yes, with the config above, I get `mvollmer/bots-github-push-tool` for both @{u} and  @{push}.